dev-util/clion/clion-2020.3.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop wrapper
+
+DESCRIPTION="A complete toolset for C and C++ development"
+HOMEPAGE="https://www.jetbrains.com/clion"
+SRC_URI="https://download.jetbrains.com/cpp/CLion-${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )
+	Apache-1.1 Apache-2.0 BSD BSD-2 CC0-1.0 CDDL-1.1 CPL-0.5 CPL-1.0
+	EPL-1.0 EPL-2.0 GPL-2 GPL-2-with-classpath-exception GPL-3 ISC JDOM
+	LGPL-2.1+ LGPL-3 MIT MPL-1.0 MPL-1.1 OFL public-domain PSF-2 UoI-NCSA ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+RESTRICT="bindist mirror splitdebug"
+
+BDEPEND="dev-util/patchelf"
+
+# RDEPENDS may cause false positives in repoman.
+# clion requires cmake and gdb at runtime to build and debug C/C++ projects
+RDEPEND="
+	dev-libs/libdbusmenu
+	dev-util/cmake
+	sys-devel/gdb"
+
+QA_PREBUILT="opt/${P}/*"
+
+src_prepare() {
+	default
+
+	local remove_me=(
+		bin/gdb/linux
+		bin/lldb/linux
+		bin/cmake
+		license/CMake*
+		lib/pty4j-native/linux/aarch64
+		lib/pty4j-native/linux/mips64el
+		lib/pty4j-native/linux/ppc64le
+	)
+
+	use amd64 || remove_me+=( bin/fsnotifier64 lib/pty4j-native/linux/x86_64)
+	use x86 || remove_me+=( bin/fsnotifier lib/pty4j-native/linux/x86)
+
+	rm -rv "${remove_me[@]}" || die
+
+	for file in "jbr/lib/"/{libjcef.so,jcef_helper}
+	do
+		if [[ -f "${file}" ]]; then
+			patchelf --set-rpath '$ORIGIN' ${file} || die
+		fi
+	done
```

Comment:
Not sure if this is working,
```
 * Messages for package dev-util/clion-2020.3:

 * QA Notice: Unresolved soname dependencies:
 * 
 * 	/opt/clion-2020.3/jbr/lib/libawt_xawt.so: libXi.so.6 libXtst.so.6
 * 	/opt/clion-2020.3/jbr/lib/libcef.so: libXcomposite.so.1 libXcursor.so.1 libXi.so.6 libXss.so.1 libXtst.so.6 libasound.so.2 libatk-1.0.so.0 libatk-bridge-2.0.so.0 libatspi.so.0 libcups.so.2 libnss3.so libnssutil3.so libpango-1.0.so.0 libpangocairo-1.0.so.0 libsmime3.so
 * 	/opt/clion-2020.3/jbr/lib/libfontmanager.so: libfreetype.so.6
 * 	/opt/clion-2020.3/jbr/lib/libjsound.so: libasound.so.2
 * 	/opt/clion-2020.3/jbr/lib/libnewt_head.so: libXcursor.so.1 libXi.so.6
 * 
```
Or you need to add many, many more deps into `RDEPEND`. And yes, it doesn't start.
```
 >: clion 
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.

Start Failed
Internal error. Please refer to https://jb.gg/ide/critical-startup-errors

com.intellij.ide.plugins.StartupAbortedException: UI initialization failed
	at com.intellij.idea.StartupUtil.lambda$prepareApp$3(StartupUtil.java:193)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:986)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:970)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at com.intellij.idea.StartupUtil.lambda$scheduleInitUi$9(StartupUtil.java:373)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.CompletionException: java.lang.UnsatisfiedLinkError: /opt/clion-2020.3/jbr/lib/libawt_xawt.so: libXtst.so.6: cannot open shared object file: No such file or directory
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$BiRelay.tryFire(CompletableFuture.java:1423)
	... 10 more
Caused by: java.lang.UnsatisfiedLinkError: /opt/clion-2020.3/jbr/lib/libawt_xawt.so: libXtst.so.6: cannot open shared object file: No such file or directory
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
	at java.base/java.lang.Runtime.load0(Runtime.java:768)
	at java.base/java.lang.System.load(System.java:1837)
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2648)
	at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:830)
	at java.base/java.lang.System.loadLibrary(System.java:1873)
	at java.desktop/java.awt.Toolkit$3.run(Toolkit.java:1395)
	at java.desktop/java.awt.Toolkit$3.run(Toolkit.java:1393)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.Toolkit.loadLibraries(Toolkit.java:1392)
	at java.desktop/java.awt.Toolkit.<clinit>(Toolkit.java:1430)
	at java.desktop/java.awt.EventQueue.invokeLater(EventQueue.java:1318)
	at com.intellij.idea.StartupUtil.lambda$scheduleInitUi$9(StartupUtil.java:324)
	... 7 more

-----
JRE 11.0.9+11-b1145.21 amd64 by JetBrains s.r.o.
/opt/clion-2020.3/jbr

Also, a UI exception occurred on an attempt to show the above message
java.lang.NoClassDefFoundError: Could not initialize class java.awt.Toolkit
	at java.desktop/java.awt.Component.<clinit>(Component.java:626)
	at com.intellij.idea.Main.showMessage(Main.java:270)
	at com.intellij.idea.Main.showMessage(Main.java:236)
	at com.intellij.ide.plugins.StartupAbortedException.logAndExit(StartupAbortedException.java:86)
	at com.intellij.ide.plugins.StartupAbortedException.processException(StartupAbortedException.java:34)
	at com.intellij.idea.StartupUtil.lambda$prepareApp$3(StartupUtil.java:193)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:986)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:970)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at com.intellij.idea.StartupUtil.lambda$scheduleInitUi$9(StartupUtil.java:373)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
2020-12-13 09:13:03,016 [     28]  ERROR - llij.ide.plugins.PluginManager - null 
java.lang.ExceptionInInitializerError
	at java.prefs/java.util.prefs.FileSystemPreferencesFactory.userRoot(FileSystemPreferencesFactory.java:41)
	at java.prefs/java.util.prefs.Preferences.userRoot(Preferences.java:457)
	at com.intellij.ide.Prefs.migrate(Prefs.java:123)
	at com.intellij.ide.Prefs.get(Prefs.java:24)
	at com.intellij.ide.gdpr.EndUserAgreement.getAcceptedVersion(EndUserAgreement.java:81)
	at com.intellij.ide.gdpr.EndUserAgreement$Document.isAccepted(EndUserAgreement.java:221)
	at com.intellij.idea.StartupUtil.showUserAgreementAndConsentsIfNeeded(StartupUtil.java:790)
	at com.intellij.idea.StartupUtil.startApp(StartupUtil.java:264)
	at com.intellij.idea.StartupUtil.prepareApp(StartupUtil.java:242)
	at com.intellij.ide.plugins.MainRunner.lambda$start$1(MainRunner.java:41)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Shutdown in progress
	at java.base/java.lang.ApplicationShutdownHooks.add(ApplicationShutdownHooks.java:66)
	at java.base/java.lang.Runtime.addShutdownHook(Runtime.java:216)
	at java.prefs/java.util.prefs.FileSystemPreferences$4.run(FileSystemPreferences.java:457)
	at java.prefs/java.util.prefs.FileSystemPreferences$4.run(FileSystemPreferences.java:455)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.prefs/java.util.prefs.FileSystemPreferences.<clinit>(FileSystemPreferences.java:455)
	... 11 more
2020-12-13 09:13:03,017 [     29]  ERROR - llij.ide.plugins.PluginManager - CLion 2020.3  Build #CL-203.5981.166 
2020-12-13 09:13:03,023 [     35]  ERROR - llij.ide.plugins.PluginManager - JDK: 11.0.9; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o. 
2020-12-13 09:13:03,023 [     35]  ERROR - llij.ide.plugins.PluginManager - OS: Linux 

Start Failed
Internal error. Please refer to https://jb.gg/ide/critical-startup-errors

java.lang.ExceptionInInitializerError
	at java.prefs/java.util.prefs.FileSystemPreferencesFactory.userRoot(FileSystemPreferencesFactory.java:41)
	at java.prefs/java.util.prefs.Preferences.userRoot(Preferences.java:457)
	at com.intellij.ide.Prefs.migrate(Prefs.java:123)
	at com.intellij.ide.Prefs.get(Prefs.java:24)
	at com.intellij.ide.gdpr.EndUserAgreement.getAcceptedVersion(EndUserAgreement.java:81)
	at com.intellij.ide.gdpr.EndUserAgreement$Document.isAccepted(EndUserAgreement.java:221)
	at com.intellij.idea.StartupUtil.showUserAgreementAndConsentsIfNeeded(StartupUtil.java:790)
	at com.intellij.idea.StartupUtil.startApp(StartupUtil.java:264)
	at com.intellij.idea.StartupUtil.prepareApp(StartupUtil.java:242)
	at com.intellij.ide.plugins.MainRunner.lambda$start$1(MainRunner.java:41)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Shutdown in progress
	at java.base/java.lang.ApplicationShutdownHooks.add(ApplicationShutdownHooks.java:66)
	at java.base/java.lang.Runtime.addShutdownHook(Runtime.java:216)
	at java.prefs/java.util.prefs.FileSystemPreferences$4.run(FileSystemPreferences.java:457)
	at java.prefs/java.util.prefs.FileSystemPreferences$4.run(FileSystemPreferences.java:455)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.prefs/java.util.prefs.FileSystemPreferences.<clinit>(FileSystemPreferences.java:455)
	... 11 more

-----
JRE 11.0.9+11-b1145.21 amd64 by JetBrains s.r.o.
/opt/clion-2020.3/jbr

Also, a UI exception occurred on an attempt to show the above message
java.lang.NoClassDefFoundError: Could not initialize class javax.swing.JTextPane
	at com.intellij.idea.Main.showMessage(Main.java:270)
	at com.intellij.idea.Main.showMessage(Main.java:236)
	at com.intellij.ide.plugins.StartupAbortedException.logAndExit(StartupAbortedException.java:86)
	at com.intellij.ide.plugins.StartupAbortedException.processException(StartupAbortedException.java:34)
	at com.intellij.ide.plugins.MainRunner.lambda$start$1(MainRunner.java:44)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
A bit weird they add this many deps on a minor version bump, unless they use quarters (should be 2020.4 now though?)

Reply:
> Or you need to add many, many more deps into RDEPEND. And yes, it doesn't start.

I don't know, why I only got two `QA Notice: Unresolved soname dependencies`, which I solved with `patchelf`. On my machine Clion works very well without any problems, or unresolved QA notices.
Do you have any suggestions how I can reproduce your behavior?

> ...unless they use quarters (should be 2020.4 now though?)

I think they don't use quarters. It's simply the 3rd major release in 2020.

> A bit weird they add this many deps on a minor version bump...

In the past we had the use flag `custom-jdk` for clion. 
If set, we installed the bundled jbr (JetBrainsRuntime), which is a modified open-jdk. 
If unset we've unbundled the jbr, and Clion had used the system java-vm.
Since 2020.3 Clion (and the other JetBrains IDEs) require JDK 11.
We don't have oracles-jdk in portage anymore, icedtea only supports JDK 8, and it's explicitly recommended to not use openjdk-11 as system java-vm at the moment.
That is why I removed the `custom-jdk` use flag with this version, and install the jbr per default. Maybe this is the reason, why it looks like a big change.

Reply:
You can reproduce the issues by emerging this clion on an empty stage3 chroot/container, with these settings:
https://wiki.gentoo.org/wiki/Package_testing#make.conf

I believe you can solve this issue simply by adding the missing sonames to RDEPEND. Sorry I didn't try it earlier, but should be as simple as that. The sonames listed above are pretty easily figurarable as well, you can use `equery b` in your own system, or let me know if you can't find which package provides a certain library.

Reply:
done